src/runtime/metal/metal_common.h

```diff
@@ -42,9 +42,34 @@
 
 #include "../workspace_pool.h"
 
+#define AUTORELEASEPOOL tvm::runtime::metal::AutoReleasePoolWrapper::GetInstance() << [&]()
```

Comment:
some documentationc comment about this macro would very helpful. A better naming is also good, how about AUTORELEASEPOOL_WITH_EXCEPTION_HANDLING

Reply:
I added documentation comments about macro and `AutoReleasePoolWrapper`.